paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -837,21 +753,21 @@ internal class PaymentSheetViewModel @Inject internal constructor(
         return listOf(target)
     }
 
-    private suspend fun awaitPaymentResult(): InternalPaymentResult? {
-        val shouldAwait = savedStateHandle.remove<Boolean>(AwaitingPaymentResultKey) ?: false
+    private suspend fun awaitIntentResult(): IntentConfirmationHandler.Result? {
```

Comment:
Should this be moved into the handler?

Reply:
This has been pretty hard to reason about having in the handler given the logic path it takes before we await for the result. Maybe we can pair on it?